app/actions/local/category.ts

```diff
@@ -45,6 +46,18 @@ export const storeCategories = async (serverUrl: string, categories: CategoryWit
     const models = await Promise.all(modelPromises);
     const flattenedModels = models.flat() as Model[];
 
+    if (prune) {
+        const remoteCategoryIds = categories.map((cat) => cat.id);
+        const localCategories = await queryCategoriesByTeamId(database, categories[0].team_id);
```

Comment:
Can we be sure there is at least one element on the categories?
Can we be sure that all the categories will always be from the same team?

Reply:
agree with this

Reply:
- `Category.id` is unique (across teams), plus `CategoryWithChannels` has a `team_id` - we don't need to worry about team here
- I've added a length check alongside prune

Reply:
I've added a `prune && categories.length` to double check.

Regarding team(s) handling; `fetchCategories` only returns categories for a given team (hence using `categories[0].team_id`, but you can pass in `categories` from a different source; potentially even all categories across all teams - so I've added a method to get the unique `team_ids` from the categories array, and get all the categories already on device that match those teams, and update accordingly. This does make `prune` all powerful, so we have to be careful when using it.